extensions/wikia/HTTPSOptIn/HTTPSOptInHooks.class.php

```diff
@@ -26,13 +26,14 @@ public static function onGetPreferences( User $user, array &$preferences ): bool
 	}
 
 	public static function onMercuryWikiVariables( array &$wikiVariables ): bool {
+		global $wgDisableHTTPSDowngrade;
 		$basePath = $wikiVariables['basePath'];
 		$user = RequestContext::getMain()->getUser();
 		if ( startsWith( $basePath, 'http://' ) && self::userAllowedHTTPS( $user ) ) {
 			$wikiVariables['basePath'] = preg_replace( '/^http:\/\//', 'https://', $basePath );
 		} elseif ( startsWith( $basePath, 'https://' ) &&
 			!self::userAllowedHTTPS( $user ) &&
-			!self::disableHTTPSDowngrade()
+			!$wgDisableHTTPSDowngrade
```

Comment:
We should either check `empty( $wgDisableHTTPSDowngrade )` here, since there isn't a default value for `wgDisableHTTPSDowngrade`, or set `$wgDisableHTTPSDowngrade = false;` as the default value in our config.

Reply:
@Grunny fix in 53af388e60f